### ISSUE TITLE
User edit dialog expects username

### ISSUE THREAD

**Issue by bossenti** [apache/streampipes#2681]
### Apache StreamPipes version

dev (current development state)

### Affected StreamPipes components

UI

### What happened?

When editing a user in the user management, the dialog expects the user name to be provided.

![image](https://github.com/apache/streampipes/assets/50115603/8cf3ebab-f0e5-4795-9b2a-041a56d88248)

This is an issue since the username is never present when the user was created through self-registration.

### Are you willing to submit a PR?

No, I don't want to submit a PR.

**Comment by muyangye** [comment]
Hey @bossenti! This issue also occurs when some other admins want to edit the initial admin admin@streampipes.apache.org. I believe the expected behavior is that the username field is not required right? If so it's just a one-line change. I have found the line but since this change is so simple, maybe we should label it as a good-first-issue so that first time contributors can have a try? If nobody claimes it for say 2 months, I'll push the fix.

**Comment by bossenti** [comment]
Good idea @muyangye, let's do it like that
In addition, we can think of if we want to keep the property `Full Name` both in the UI and the data model. AFAIK, it is not used currently

**Comment by muyangye** [comment]
In my opinion, we should probably keep it despite it's not currently used in code because users might want to know that attribute. For example, an assembly line may have multiple industrial/manufactural engineers who use StreamPipes and they/their admin might be interested in knowing who is who. What do you think?

**Comment by bossenti** [comment]
Yes, I agree. In mid-/longterm we might want to use it, especially if integrating SSO somewhen

**Comment by Hitaishi08** [comment]
Hello @bossenti
I’m interested in working on this issue and would like to take it up.
Kindly assign it to me if it’s available.

**Comment by muyangye** [comment]
I forgot about this issue, @Hitaishi08 feel free to just take over. Looks that I can assign so I assigned to you. Lmk if you need anything. Thanks a lot for your help🙂!

**Comment by AliRana30** [comment]
Hi @bossenti @muyangye — I noticed this issue is still open and the previous assignee hasn't been active for about 2 months. I've looked into it and based on the earlier discussion, the fix involves removing the **required** constraint from the username field in the user edit dialog, since users created via self-registration don't have a username stored.
I'm interested in working on this. Could you re-assign this to me, or confirm it's still available?

**Comment by muyangye** [comment]
@AliRana30 Sorry for the late reply, assigned
